packages/console/src/components/Search/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { FormEventHandler, KeyboardEventHandler, useState } from 'react';
+
+import SearchIcon from '@/icons/Search';
+
+import Button from '../Button';
+import TextInput from '../TextInput';
+import * as styles from './index.module.scss';
+
+type Props = {
+  defaultValue?: string;
+  onSearch?: (value: string) => void;
+};
+
+const Search = ({ defaultValue = '', onSearch }: Props) => {
+  const [inputValue, setInputValue] = useState<string>(defaultValue);
+
+  const handleSearchKeyPress: KeyboardEventHandler<HTMLInputElement> = (event) => {
+    if (event.key === 'Enter') {
+      onSearch?.(inputValue);
+    }
+  };
```

Comment:
Add input debounce

Reply:
We only listen to "Enter" key, so I think "debounce" is not necessary here.

Reply:
Yes, but what if someone repeatly hitting the enter key? Well, I agree it's non-blocking thing. Let's leave it as-is.

Reply:
Good call, I created LOG-2049 to track this edge case.